fdbserver/workloads/AtomicRestore.actor.cpp

```diff
@@ -42,7 +43,10 @@ struct AtomicRestoreWorkload : TestWorkload {
 		startAfter = getOption(options, "startAfter"_sr, 10.0);
 		restoreAfter = getOption(options, "restoreAfter"_sr, 20.0);
 		fastRestore = getOption(options, "fastRestore"_sr, false);
-		if (!fastRestore) {
+		// TODO: Currently atomic restore does not work well with encryption since we cannot
+		// seperate system key restore from the normal key restore, once we can support this (or no longer need to
```

Comment:
The first atomic restore (to restore the system keys) will complete the current backup. Then once the second atomic restore runs (to restore the user keys) we will hit this condition: https://github.com/apple/foundationdb/blob/main/fdbclient/FileBackupAgent.actor.cpp#L5940 since there is currently no backup running (the status is: `STATE_COMPLETED`). This prevents us from running them in sequence since the second will never finish

Reply:
I see, I wonder if the implementation for the sequenced backups could be put inside of atomic restore (so rather than call it twice, you call it once and let it run two restores internally).

Reply:
I tried this and it messes up the restore. I'm not entirely familiar with this code to know exactly why this happens but it might have to do with the restore locking semantics. I can investigate this further but for the time being i'd rather have this as a todo so it can unblock other work and I can come back to this.

Reply:
I added the following code:
`state Standalone<VectorRef<KeyRangeRef>> restoreRange;
			state bool containsSystemKeys = false;
			for (auto r : ranges) {
				if (!r.intersects(getSystemBackupRanges())) {
					restoreRange.push_back_deep(restoreRange.arena(), r);
				} else {
					containsSystemKeys = true;
				}
			}
			if (containsSystemKeys) {
				wait(success(restore(backupAgent,
				                     cx,
				                     cx,
				                     tagName,
				                     KeyRef(bc->getURL()),
				                     bc->getProxy(),
				                     getSystemBackupRanges(),
				                     WaitForComplete::True,
				                     ::invalidVersion,
				                     Verbose::True,
				                     addPrefix,
				                     removePrefix,
				                     LockDB::True,
				                     OnlyApplyMutationLogs::False,
				                     InconsistentSnapshotOnly::False,
				                     ::invalidVersion,
				                     {},
				                     randomUid)));
			}`
However when the restore happens the ApiWorkload throws a bunch of Sev 40s indicating that the restore didn't happen properly